io/key/key.go

```diff
@@ -64,6 +65,23 @@ type EditEvent struct {
 	Text string
 }
 
+// InputMode changes the on-screen-keyboard type. That hints the
+// type of data that might be entered by the user.
+type InputMode uint8
+
+const (
+	// KeyboardDefault is the default keyboard (alphanumeric).
+	KeyboardDefault InputMode = iota
+	// KeyboardNumeric is an keyboard containing 0-9 Digits and decimal separator (`.` or `,`).
+	KeyboardNumeric
+	// KeyboardEmail is an keyboard containing "@", ".com" and other optimizations.
+	KeyboardEmail
+	// KeyboardUrl is an keyboard containing "/", ".com" and other optimizations.
+	KeyboardUrl
+	// KeyboardTelephone is an keyboard containing 0-9 Digits and the pound (#) and the asterisk (*).
+	KeyboardTelephone
```

Comment:
It's weird to have a common prefix `Keyboard` that doesn't match the type name. I suggest `Hint` (or `Mode` if we go with your original name)?

If so, we should change the comments to refer to the mode rather than a keyboard. For example:

// HintURL hints that URL input is expected. It may activate shortcuts for common URL fragments such as "/" and ".com".